MonoGame.Framework/Graphics/SpriteBatch.cs

```diff
@@ -847,6 +847,192 @@ public unsafe void DrawString (SpriteFont spriteFont, string text, Vector2 posit
+                {
+                    origin.Y *= -1;
+                    flipAdjustment.Y = spriteFont.LineSpacing - size.Y;
+                }
+            }
+
+            Matrix transformation = Matrix.Identity;
+            float cos = 0, sin = 0;
+            if (rotation == 0)
+            {
+                transformation.M11 = (flippedHorz ? -scale.X : scale.X);
+                transformation.M22 = (flippedVert ? -scale.Y : scale.Y);
+                transformation.M41 = ((flipAdjustment.X - origin.X) * transformation.M11) + position.X;
+                transformation.M42 = ((flipAdjustment.Y - origin.Y) * transformation.M22) + position.Y;
+            }
+            else
+            {
+                cos = MathF.Cos(rotation);
+                sin = MathF.Sin(rotation);
+                transformation.M11 = (flippedHorz ? -scale.X : scale.X) * cos;
+                transformation.M12 = (flippedHorz ? -scale.X : scale.X) * sin;
+                transformation.M21 = (flippedVert ? -scale.Y : scale.Y) * (-sin);
+                transformation.M22 = (flippedVert ? -scale.Y : scale.Y) * cos;
+                transformation.M41 = (((flipAdjustment.X - origin.X) * transformation.M11) + (flipAdjustment.Y - origin.Y) * transformation.M21) + position.X;
+                transformation.M42 = (((flipAdjustment.X - origin.X) * transformation.M12) + (flipAdjustment.Y - origin.Y) * transformation.M22) + position.Y;
+            }
+
+            var offset = Vector2.Zero;
+            var firstGlyphOfLine = true;
+
+            fixed (SpriteFont.Glyph* pGlyphs = spriteFont.Glyphs)
+                for (var i = 0; i < text.Length; ++i)
+                {
+                    var c = text[i];
+
+                    if (c == '\r')
+                        continue;
+
+                    if (c == '\n')
+                    {
+                        offset.X = 0;
+                        offset.Y += spriteFont.LineSpacing;
+                        firstGlyphOfLine = true;
+                        continue;
+                    }
+
+                    var currentGlyphIndex = spriteFont.GetGlyphIndexOrDefault(c);
+                    var pCurrentGlyph = pGlyphs + currentGlyphIndex;
+
+                    // The first character on a line might have a negative left side bearing.
+                    // In this scenario, SpriteBatch/SpriteFont normally offset the text to the right,
+                    //  so that text does not hang off the left side of its rectangle.
+                    if (firstGlyphOfLine)
+                    {
+                        offset.X = Math.Max(pCurrentGlyph->LeftSideBearing, 0);
+                        firstGlyphOfLine = false;
+                    }
+                    else
+                    {
+                        offset.X += spriteFont.Spacing + pCurrentGlyph->LeftSideBearing;
+                    }
+
+                    var p = offset;
+
+                    if (rtl ^ flippedHorz)
```

Comment:
This DrawString method appears to be the same as the one above with the exception of the extra argument and this line of code. Unless I'm missing something this just adds an extra gap between the characters and makes the rendered text look broken. I don't fully understand RTL text rendering but I don't think this will achieve it.

Reply:
You were right, it doesn't work. What I was trying to do is basically hijack the logic that prints horizontally flipped strings and use it to print rtl, but I misunderstood the way it works and without testing I didn't notice it didn't work as I intended.